presto-geospatial/src/main/java/com/facebook/presto/plugin/geospatial/GeoFunctions.java

```diff
@@ -198,6 +198,29 @@ public static Slice stCentroid(@SqlType(GEOMETRY_TYPE_NAME) Slice input)
         return serialize(createFromEsriGeometry(centroid, geometry.getEsriSpatialReference()));
     }
 
+    @Description("Returns the minimum convex geometry that encloses all input geometries")
+    @ScalarFunction("ST_ConvexHull")
+    @SqlType(GEOMETRY_TYPE_NAME)
+    public static Slice stConvexHull(@SqlType(GEOMETRY_TYPE_NAME) Slice input)
+    {
+        OGCGeometry geometry = deserialize(input);
+        validateType("ST_ConvexHull", geometry, EnumSet.of(POINT, MULTI_POINT, LINE_STRING, MULTI_LINE_STRING, POLYGON, MULTI_POLYGON));
+        if (geometry.isEmpty()) {
+            return input;
+        }
+        OGCGeometry convexHull = geometry.convexHull();
+        long pointCount = getOGCGeometryPointCount(convexHull);
+        if (pointCount == 0) {
+            return serialize(getSinglePointFromGeometry(geometry));
```

Comment:
It would be more efficient to handle points explicitly and using `isEmpty()` check in place of `pointCount == 0`. I'm explaining why in a comment attached to `getOGCGeometryPointCount`.

It would also be helpful to explain why these workarounds are needed and provide pointers to ESRI issues.

```
if (geometry.isEmpty()) {
   return input;
}

if (GeometryType.getForEsriGeometryType(geometry.geometryType()) == POINT)  {
   return input;
}

OGCGeometry convexHull = geometry.convexHull();
if (convexHull.isEmpty()) {
   // This happens for a single-point multi-point because of a bug in ESRI library - https://github.com/Esri/geometry-api-java/issues/172
   return serialize(createFromEsriGeometry(((MultiVertexGeometry) geometry.getEsriGeometry()).getPoint(0), null));
}

if (convexHull instanceOf MultiVertexGeometry) {
    MultiVertexGeometry multiVertex = (MultiVertexGeometry) convexHull;
    if (multiVertex. getPointCount() == 2) {
       // This happens when all points of the input geometry are on the same line because of a bug in ESRI library - https://github.com/Esri/geometry-api-java/issues/172
        ...
    }
}

return serialize(convexHull);
```